frontend/src/util/err.ts

```diff
@@ -0,0 +1,41 @@
+// Make sure that the parameter is `never`. This is particularly useful for
+// enum-like types where you want to make sure you handled all cases. For
+// example:
+//
+// ```
+// type Foo = "anna" | "bob";
+// const x = "anna" as Foo;
+//
+// switch x {
+//     case "anna": ...; break;
+//     case "bob": ...; break;
+//     default: assertNever(x);
+// }
+// ```
+//
+// If you add another variant to that type, the switch statement will fail to
+// compile. This is what we want in most cases! That way, we are notified of all
+// places which we might need to change after adding a new variant.
+export const assertNever = (n: never): never =>
+    bug("`assertNever` call was reached, that's a soundness hole in the typesystem :(");
+
+// A custom error type that represents bugs: errors that are not expected and
+// that cannot be handled. They are caused by a bug in our code and not by the
+// "world" (e.g. any input). Use the helper functions below to throw this error.
+export class Bug extends Error {
+    public constructor(msg: string) {
```

Comment:
I assume the explicit `public` is because of a Lint? Do we agree with that?

Reply:
Yes because of lint. And ... yes I guess I can live with the lint? I don't know what visibility it would have without keyword :D (tho that's a weak argument of course). But yeah, do you dislike explicit keywords here?

Reply:
Random thoughts:

* I don't like Java's "lets have 5 keywords in front of every method name, 7 if you want to do fancy shit"
  * In JS there aren't that many modifiers, of course, but this still reminds me of that
* The default is `public`
* Will we even make use of TS's visibility system? I.e. will we ever have `private` stuff? **Or** will we use `#privateFields`? This is not a rhetorical question, I genuinely don't know. Maybe even both, since there are slight differences. However, if we decide to go exclusively or mostly with the latter, I think this lint should go, since the default does what we want.

Reply:
Ok, I guess this needs some more research and decision making. I will create an issue. This doesn't have to block this PR of course.

Reply:
Said issue: #73